packages/realm-server/tests/realm-endpoints/lint-test.ts

```diff
@@ -42,130 +54,821 @@ module(`realm-endpoints/${basename(__filename)}`, function () {
+          );
+
+          // Check that formatting actually improves the code structure
+          // The input is minified, so formatting should add whitespace
+          const hasProperSpacing =
+            result.includes('import {') ||
+            result.includes("from '") ||
+            result.includes('export ');
+          assert.ok(
+            hasProperSpacing,
+            `Formatted output has proper spacing. Input: ${input.length} chars, Output: ${result.length} chars. Result: ${result.substring(0, 100)}...`,
+          );
+        } catch (error) {
+          assert.ok(
+            false,
+            `Basic prettier formatting failed: ${(error as Error).message}`,
+          );
+        }
+      });
+
+      test('performance benchmarking works', async function (assert) {
+        const benchmark = await benchmarkOperation(
+          'test-operation',
+          () => new Promise((resolve) => setTimeout(resolve, 1)),
+          10,
+        );
+
+        assert.ok(benchmark.duration > 0, 'Benchmark should record duration');
+        assert.strictEqual(
+          benchmark.iterations,
+          10,
+          'Should run correct number of iterations',
+        );
+        assert.ok(benchmark.averageTime > 0, 'Should calculate average time');
+        assert.ok(
+          benchmark.maxTime >= benchmark.minTime,
+          'Max time should be >= min time',
+        );
+      });
     });
 
-    test('user can do a lint with missing template invokable fix', async function (assert) {
-      let response = await request
-        .post('/_lint')
-        .set(
-          'Authorization',
-          `Bearer ${createJWT(testRealm, 'john', ['read', 'write'])}`,
-        )
-        .set('X-HTTP-Method-Override', 'QUERY')
-        .set('Accept', 'application/json')
-        .send(`import MyComponent from 'somewhere';
-<template>
-  <MyComponent @flag={{eq 1 1}} />
-</template>
-`);
+    module('Backward Compatibility Tests', function () {
+      test('existing lint consumers continue to work', async function (assert) {
+        const compatibility = await testBackwardCompatibility(
+          request,
+          testRealm,
+          createJWT,
+        );
 
-      assert.strictEqual(response.status, 200, 'HTTP 200 status');
-      let responseJson = JSON.parse(response.text);
-      assert.strictEqual(
-        responseJson.output,
-        `import { eq } from '@cardstack/boxel-ui/helpers';
-import MyComponent from 'somewhere';
-<template>
-  <MyComponent @flag={{eq 1 1}} />
-</template>
-`,
-      );
+        assert.ok(
+          compatibility.isCompatible,
+          `Backward compatibility issues: ${compatibility.errors.join(', ')}`,
+        );
+        assert.ok(compatibility.results.length > 0, 'Should have test results');
+
+        for (const result of compatibility.results) {
+          assert.ok(
+            result.success,
+            `Test case '${result.name}' should succeed`,
+          );
+        }
```

Comment:
This is another case where I don’t understand the use of a helper function, it makes it harder to follow to bury what’s being tested in a one-use external function